OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -99,6 +113,17 @@ public bool CanTarget(Actor self, Target target, List<Actor> othersAtTarget, ref
 				if (self.World.Map.Contains(location))
 				{
 					cursor = "ability";
+
+					// Set also PrimaryBuilding with Ctrl and only this one building of the same type selected
+					if (modifiers.HasModifier(TargetModifiers.ForceAttack))
+					{
+						var selfName = self.Info.Name;
+						if (self.World.Selection.Actors.Count(a => a.Info.Name == selfName) == 1)
+						{
+							SetPrimaryBuilding = true;
+						}
+					}
+
```

Comment:
I was afraid of this also, wasn't sure. It could be set always to False in CanTarget for future, but isn't needed currently. RallyPoint.Orders always creates new instance of RallyPointOrderTargeter, lately CanTarget is called, it is never reused.

Thanks for all your comments.